packages/x-data-grid-pro/src/hooks/features/dataSource/utils.ts

```diff
@@ -112,3 +126,78 @@ export class NestedDataManager {
 
   public getActiveRequestsCount = () => this.pendingRequests.size + this.queuedRequests.size;
 }
+
+/**
+ * Provides better cache hit rate by splitting the data into smaller chunks
+ * Splits the data into smaller chunks to be stored in the cache
+ * Merges multiple cache entries into a single response
+ */
+export class CacheChunkManager {
+  private chunkSize: number;
+
+  constructor(config: GridDataSourceCacheChunkManagerConfig) {
+    this.chunkSize = config.chunkSize;
+  }
+
+  public getCacheKeys = (key: GridGetRowsParams) => {
+    if (this.chunkSize < 1 || typeof key.start !== 'number') {
+      return [key];
+    }
+
+    // split the range into chunks
+    const chunkedKeys: GridGetRowsParams[] = [];
+    for (let i = key.start; i < key.end; i += this.chunkSize) {
+      const end = Math.min(i + this.chunkSize - 1, key.end);
+      chunkedKeys.push({ ...key, start: i, end });
+    }
+
+    return chunkedKeys;
+  };
+
+  public splitResponse = (key: GridGetRowsParams, response: GridGetRowsResponse) => {
+    const cacheKeys = this.getCacheKeys(key);
+    const responses = new Map<GridGetRowsParams, GridGetRowsResponse>();
+    cacheKeys.forEach((chunkKey) => {
+      const isLastChunk = chunkKey.end === key.end;
+      const responseSlice: GridGetRowsResponse = {
+        ...response,
+        pageInfo: {
+          ...response.pageInfo,
+          // If the original response had page info, update that information for all but last chunk and keep the original value for the last chunk
+          hasNextPage:
+            (response.pageInfo?.hasNextPage !== undefined && !isLastChunk) ||
+            response.pageInfo?.hasNextPage,
+          nextCursor:
+            response.pageInfo?.nextCursor !== undefined && !isLastChunk
+              ? response.rows[chunkKey.end + 1].id
+              : response.pageInfo?.nextCursor,
+        },
+        rows:
+          typeof chunkKey.start !== 'number' || typeof key.start !== 'number'
+            ? response.rows
+            : response.rows.slice(chunkKey.start - key.start, chunkKey.end - key.start + 1),
+      };
+
+      responses.set(chunkKey, responseSlice);
+    });
+
+    return responses;
+  };
+
+  static mergeResponses = (responses: GridGetRowsResponse[]): GridGetRowsResponse => {
+    if (responses.length === 1) {
+      return responses[0];
+    }
+
+    return responses.reduce(
+      (acc, response) => {
+        return {
+          rows: [...acc.rows, ...response.rows],
+          rowCount: response.rowCount,
+          pageInfo: response.pageInfo,
+        };
+      },
+      { rows: [], rowCount: 0, pageInfo: {} },
+    );
```

Comment:
Optional optimization.
```suggestion
    return responses.reduce(
      (acc, response) => {
        acc.rows.push(...response.rows);
        return acc;
      },
      {
        rows: [],
        rowCount: responses[responses.length - 1].rowCount,
        pageInfo: responses[responses.length - 1].pageInfo,
      },
    );
```

Reply:
Already done in one of the previous commits :)

Reply:
It seems like you still need to push it, right? I can't see in the latest commit on the PR.